host/hackrf-tools/src/hackrf_sweep.c

```diff
@@ -741,6 +741,8 @@ int main(int argc, char** argv) {
 
 	gettimeofday(&time_now, NULL);
 	time_diff = TimevalDiff(&time_now, &t_start);
+	if((sweep_rate == 0) && (time_diff > 0))
```

Comment:
There is a chance that this check for ```sweep_rate == 0``` may be checking against an uninitialized instance of sweep_rate because of a preexisting issue (see #869 and #870) . I would also recommend initializing sweep_rate to 0 on line 401.

Reply:
Thanks for spotting this.
Pushed a change.